libp2p/src/tutorials/ping.rs

```diff
@@ -71,50 +71,39 @@
 //! derived from their public key. Now, replace the contents of main.rs by:
 //!
 //! ```rust
-//! use libp2p::{identity, PeerId};
 //! use std::error::Error;
 //!
 //! #[async_std::main]
 //! async fn main() -> Result<(), Box<dyn Error>> {
-//!     let local_key = identity::Keypair::generate_ed25519();
-//!     let local_peer_id = PeerId::from(local_key.public());
-//!     println!("Local peer id: {local_peer_id:?}");
+//!     let mut swarm = libp2p::SwarmBuilder::with_new_identity();
 //!
 //!     Ok(())
 //! }
 //! ```
 //!
-//! Go ahead and build and run the above code with: `cargo run`. A unique
-//! [`PeerId`](crate::PeerId) should be displayed.
+//! Go ahead and build and run the above code with: `cargo run`. Nothing happening thus far.
 //!
 //! ## Transport
 //!
-//! Next up we need to construct a transport. A transport in libp2p provides
-//! connection-oriented communication channels (e.g. TCP) as well as upgrades
-//! on top of those like authentication and encryption protocols. Technically,
-//! a libp2p transport is anything that implements the [`Transport`] trait.
-//!
-//! Instead of constructing a transport ourselves for this tutorial, we use the
-//! convenience function [`development_transport`](crate::development_transport)
-//! that creates a TCP transport with [`noise`](crate::noise) for authenticated
-//! encryption.
-//!
-//! Furthermore, [`development_transport`] builds a multiplexed transport,
-//! whereby multiple logical substreams can coexist on the same underlying (TCP)
-//! connection. For further details on substream multiplexing, take a look at
-//! [`crate::core::muxing`] and [`yamux`](crate::yamux).
+//! Next up we need to construct a transport. Each transport in libp2p provides encrypted streams.
+//! E.g. combining TCP to establish connections, TLS to encrypt these connections and Yamux to run
+//! one or more streams on a connection. Another libp2p transport is QUIC, providing encrypted
+//! streams out-of-the-box. We will stick to TCP for now. Each of these implement the [`Transport`]
+//! trait.
 //!
 //! ```rust
 //! use libp2p::{identity, PeerId};
 //! use std::error::Error;
 //!
 //! #[async_std::main]
 //! async fn main() -> Result<(), Box<dyn Error>> {
-//!     let local_key = identity::Keypair::generate_ed25519();
-//!     let local_peer_id = PeerId::from(local_key.public());
-//!     println!("Local peer id: {local_peer_id:?}");
-//!
-//!     let transport = libp2p::development_transport(local_key).await?;
+//!     let mut swarm = libp2p::SwarmBuilder::with_new_identity()
+//!         .with_async_std()
+//!         .with_tcp(
+//!             Default::default(),
```

Comment:
Let's not use `Default::default` here please.

Reply:
8c12ec51b